### ISSUE TITLE
Crash when pulling trigger in Packaged Game Only

### ISSUE THREAD

**Issue by BlakeLivSim** [ErikHedberg/Terminal-Ballistics-Example-Project#7]
VCRUNTIME140
MyProject_Win64_DebugGame!FTBBullet::FTBBullet()
MyProject_Win64_DebugGame!FTBBullet::FTBBullet() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Core\TBBullets.cpp:309]
MyProject_Win64_DebugGame!FTBBullet::CreatePtrFromBulletDataAsset() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Core\TBBullets.cpp:373]
MyProject_Win64_DebugGame!UTerminalBallisticsSubsystem::CreateBulletSimDataFromDataAsset() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Subsystems\TerminalBallisticsSubsystem.cpp:437]
MyProject_Win64_DebugGame!UTerminalBallisticsStatics::AddAndFireBulletWithCallbacks() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Core\TBStatics.cpp:1575]
MyProject_Win64_DebugGame!UTerminalBallisticsStatics::execAddAndFireBulletWithCallbacks() [d:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Intermediate\Build\Win64\UnrealGame\Inc\TerminalBallistics\UHT\TBStatics.gen.cpp:2

When I pull the trigger, it crashes with the previous list. 

When I attach VS to the packaged debug exe, and I set a breakpoint on Line 282 of TBBullets.cpp, it breaks on Line 309, "BulletName = bullet.BulletName;" - can you advise.  I"m using the 9 MM data asset, I'll try a packaged with a different version.  AFAIK I have not made any changes to the TB Bullet Data Assets.

**Comment by BlakeLivSim** [comment]
Additional info, might not be related, getting warning for the following - VerifyImport: Failed to load package for import object 'Package /TerminalBallistics/PhysicalMaterials/Tile_TB'
 Failed to load package for import object 'Package /TerminalBallistics/PhysicalMaterials/PassThrough'

**Comment by ErikHedberg** [comment]
Not sure what those warning are since neither of those are in the "PhysicalMaterials" directory, but that seems like an odd place to crash. What was the actual error that caused the crash?

**Comment by BlakeLivSim** [comment]
It seems to be some null data asset.  I'm doublechecking that each Launch Parameter has an owner but I believe the BP node has a self functionality there, which shouldn't break anything AFAIK.  Specifically though that line I gave you in TBBullets.cpp is the one the breakpoint fires.
________________________________

**Comment by ErikHedberg** [comment]
Yeah, I'm guessing it is a nullptr access issue, but I don't know what it is trying to access exactly.

```cpp
if (DataAsset->bUseExistingBullet && !DataAsset->bCustomizePreset) // Data asset isn't invalid, it is being accessed here
{
	if (const auto FoundBulletProperties = TB::Bullets::BulletTypes::Get().GetMappings().Find(DataAsset->PresetBullets))
	{
		FTBBullet bullet = *FoundBulletProperties; // Dereferencing a pointer, but the validity of it was checked above
Crash ----->	BulletName = bullet.BulletName;
```

**Comment by BlakeLivSim** [comment]
I've checked every instance of LaunchParameters when I spawn bullets and something (even a reference to self) is plugged into the owner, it's not that.

Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0xffffffffffffffff

VCRUNTIME140
MyProject_Win64_DebugGame!FTBBullet::FTBBullet()
MyProject_Win64_DebugGame!FTBBullet::FTBBullet() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Core\TBBullets.cpp:309]
MyProject_Win64_DebugGame!FTBBullet::CreatePtrFromBulletDataAsset() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Core\TBBullets.cpp:373]
MyProject_Win64_DebugGame!UTerminalBallisticsSubsystem::CreateBulletSimDataFromDataAsset() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Subsystems\TerminalBallisticsSubsystem.cpp:437]
MyProject_Win64_DebugGame!UTerminalBallisticsStatics::AddAndFireBulletWithCallbacks() [D:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Source\TerminalBallistics\Private\Core\TBStatics.cpp:1575]
MyProject_Win64_DebugGame!UTerminalBallisticsStatics::execAddAndFireBulletWithCallbacks() [d:\build\U5M-Marketplace\Sync\LocalBuilds\PluginTemp\HostProject\Plugins\TerminalBallistics\Intermediate\Build\Win64\UnrealGame\Inc\TerminalBallistics\UHT\TBStatics.gen.cpp:242]
MyProject_Win64_DebugGame!UObject::execCallMathFunction()
MyProject_Win64_DebugGame!UObject::execLet()
MyProject_Win64_DebugGame!ProcessLocalScriptFunction()
MyProject_Win64_DebugGame!ProcessScriptFunction<void (__cdecl*)(UObject * __ptr64,FFrame & __ptr64,void * __ptr64)>()
MyProject_Win64_DebugGame!ProcessLocalFunction()
MyProject_Win64_DebugGame!ProcessLocalScriptFunction()
MyProject_Win64_DebugGame!ProcessScriptFunction<void (__cdecl*)(UObject * __ptr64,FFrame & __ptr64,void * __ptr64)>()
MyProject_Win64_DebugGame!ProcessLocalFunction()
MyProject_Win64_DebugGame!ProcessLocalScriptFunction()


Is there anyway we could drop on a quick call, I could show you what I'm doing and you might spot the issue - Here is my main way to fire-  Am I doing something wrong here?

[cid:9b1b6b26-d01b-4825-a84a-7528fb8b6ece]
________________________________

**Comment by BlakeLivSim** [comment]
Additionally, here is the Data Asset for the Bullet - I tried packaging with the 45 ACP instead but that didn't change the crash.

[cid:b6d4fdb2-6145-4b19-91c3-59425b2c3f18]
________________________________

**Comment by ErikHedberg** [comment]
We can certainly jump in a call, feel free to send me a DM on discord. My username is "Xor"

**Comment by BlakeLivSim** [comment]
Great! Sent an invite - name should be Blakestr

________________________________
